### PR TITLE
remove python2, do not lower case

### PR DESCRIPTION
**Pull request by jedwards4b**
[ 50 character, one line summary ]
Do not use lower() on names and paths - these are case specific and so case needs to be preserved.

[ Description of the changes in this commit. It should be enough
  information for someone not following this development to understand. 
  Lines should be wrapped at about 72 characters. ]
Remove lower() function so that case of info in externals file is preserved.   Remove python2 specific code, 
python2 is no longer supported. 

User interface changes?:  No
[ If yes, describe what changed, and steps taken to ensure backward compatibilty ]

Fixes: [Github issue #s] And brief description of each issue.

Testing:
  test removed: none
  unit tests:all pass
  system tests:all pass
  manual testing:

### CODE REVIEW

**Review by fischer-ncar — APPROVED** [review]

**Comment by jedwards4b** [comment]
Note that all models using cime will need to update to this manage_externals after https://github.com/ESMCI/cime/pull/4162

File: checkout_externals [hunk 1]
```diff
@@ -17,8 +17,8 @@ import manic
 
 if sys.hexversion < 0x02070000:
     print(70 * '*')
-    print('ERROR: {0} requires python >= 2.7.x. '.format(sys.argv[0]))
-    print('It appears that you are running python {0}'.format(
+    print(f'ERROR: {0} requires python >= 2.7.x. '.format(sys.argv[0]))
+    print(f'It appears that you are running python {0}'.format(
```
Comment (gold2718) [inline checkout_externals]: If we are removing python2, why is the check still for python 2.7 and not 3?

Comment (jedwards4b) [reply checkout_externals]: Fixed. thank you.

File: manic/checkout.py [hunk 2]
```diff
@@ -24,11 +24,14 @@
...
-    print(70 * '*')
-    print('ERROR: {0} requires python >= 2.7.x. '.format(sys.argv[0]))
-    print('It appears that you are running python {0}'.format(
-        VERSION_SEPERATOR.join(str(x) for x in sys.version_info[0:3])))
-    print(70 * '*')
+    print(70 * "*")
+    print(f"ERROR: {0} requires python >= 2.7.x. ".format(sys.argv[0]))
+    print(
+        f"It appears that you are running python {0}".format(
+            VERSION_SEPERATOR.join(str(x) for x in sys.version_info[0:3])
```
Comment (gold2718) [inline manic/checkout.py:31]: Sorry, should have pointed out this one as well.

File: manic/checkout.py [hunk 3]
```diff
@@ -261,70 +264,114 @@ def commandline_arguments(args=None):
...
-                        help='By default, subversion will abort if a component is '
-                        'already checked out and there is no common ancestry with '
-                        'the new URL. This flag passes the "--ignore-ancestry" flag '
-                        'to the svn switch call. (This is not recommended unless '
-                        'you are sure about what you are doing.)')
+    parser.add_argument(
+        "components",
+        nargs="*",
+        help="Specific component(s) to checkout. By default, "
+        "all required externals are checked out.",
```
Comment (gold2718) [inline manic/checkout.py:282]: Why all the weird format changes?

Comment (jedwards4b) [reply manic/checkout.py:282]: black formating

**Comment by jedwards4b** [comment]
closing as stale

**Pull request closed without merging**